aten/src/ATen/native/AdaptiveMaxPooling3d.cpp

```diff
@@ -7,10 +7,15 @@
 namespace at {
 namespace meta {
 TORCH_META_FUNC(adaptive_max_pool3d) (const Tensor& input, IntArrayRef output_size) {
-  for (int64_t i = 0; i < input.ndimension(); i++) {
+  auto ndim = input.ndimension();
+  TORCH_CHECK(
+    ndim == 4 || ndim == 5,
+    "adaptive_max_pool3d(): Expected 4D or 5D tensor with optional 0-dim batch size, but got: ",
```

Comment:
nit: I think since this check is only concerned with the number of dims, "adaptive_max_pool3d(): Expected 4D or 5D tensor, but got: " is better